server/statistics/hot_peer_cache.go

```diff
@@ -116,9 +114,13 @@ func (f *hotPeerCache) CheckRegionFlow(region *core.RegionInfo, storesStats *Sto
 	byteRate := totalBytes / float64(interval)
 	keyRate := totalKeys / float64(interval)
 
+	// old region is in the front and new region is in the back
+	// which ensures it will hit the cache if moving peer or transfer leader occurs with the same replica number
+
 	var tmpItem *HotPeerStat
+	storeIDs := f.getAllStoreIDs(region)
 	for _, storeID := range storeIDs {
-		isExpired := f.isRegionExpired(region, storeID)
+		isExpired := f.isRegionExpired(region, storeID) // transfer leader or remove peer
```

Comment:
I realized that it seems we can directly update the cache instead of returning the item which needs to be updated?

Reply:
I will open another PR to merge them.

Reply:
why directly update the cache? we need to check it first for reducing heartbeat lock overload.